scalameter-core/src/main/scala/org/scalameter/utils/ClassPath.scala

```diff
@@ -8,51 +8,73 @@ import org.apache.commons.lang3.SystemUtils
 
 
 
-object ClassPath {
-  /** Returns the default classpath string.
+class ClassPath private (val paths: List[String]) extends Serializable {
+  /** Returns platform dependent classpath string.
    */
-  def default: String = {
-    extract(this.getClass.getClassLoader, sys.props("java.class.path"))
+  def mkString: String = {
+    val cp = paths.mkString(File.pathSeparator)
+    if (SystemUtils.IS_OS_WINDOWS) "\"" + cp + "\"" else cp
+  }
+
+  /** Prepends library to the classpath.
+   */
+  def ::(library: String) = {
+    ClassPath.validate(library)
+    new ClassPath(library :: paths)
+  }
+
+  override def equals(obj: Any): Boolean = obj match {
+    case cl: ClassPath => paths == cl.paths
+    case _ => false
+  }
+
+  override def hashCode(): Int = paths.hashCode()
+}
+
+object ClassPath {
+  private def validate(elem: String) = {
+    require(!elem.contains(File.pathSeparatorChar), s"Classpath library contains illegal character: ${File.pathSeparatorChar}")
```

Comment:
Right, I've checked such case now and it looks like that it must be enquoted to be recognized.